### DISCUSSION TITLE
How to add transitions/animations to e.g. table rows?

### DISCUSSION THREAD

**Discussion by brendan-morin** [themesberg/flowbite-svelte#1396] [Q&A]
Hi everyone, I'm looking for some basic ideas for how to add animations or transitions to flowbite-svelte table rows. Let's say I have a table like this:
```svelte
<Table>
  <TableHead>
    <TableHeadCell class="w-1">Time</TableHeadCell>
    <TableHeadCell class="w-1">User</TableHeadCell>
  </TableHead>
  <TableBody>
    {#each notes as note}
      <TableBodyRow>
        <TableBodyCell>{note.created}</TableBodyCell>
        <TableBodyCell>{note.user}</TableBodyCell>
      </TableBodyRow>
    {/each}
  </TableBody>
</Table>
```
And it renders something like this (the picture has some more features, but the above code should be accurate enough to get the point):
<img width="921" alt="Screenshot 2024-07-23 at 1 13 22 PM" src="https://github.com/user-attachments/assets/0bb0bb52-dad4-443c-8d3f-f543e656a1f5">

If a user inserts a new row, deletes a row, or some action triggers row reordering, I call `invalidateAll()` and it fetches new data, which should reflect the updated order, provided by the server. My goal is to animate this with some smoother transitions.

One challenge I'm running into is that svelte doesn't allow transitions or animations on components (e.g. `Transitions can only be applied to DOM elements, not components svelte(invalid-transition)`). Adding divs to wrap these components ([similar to this discussion](https://github.com/themesberg/flowbite-svelte/discussions/1207)) in the context of a table does create the desired transitions, but creates layout issues. For example, if I try to just wrap the rows like this:

```svelte
<Table>
  <TableHead>
    <TableHeadCell class="w-1">Time</TableHeadCell>
    <TableHeadCell class="w-1">User</TableHeadCell>
  </TableHead>
  <TableBody>
    {#each notes as note}
      <div animate:flip={{ duration: 300 }}>
        <div in:fade={{ duration: 300 }} out:slide={{ duration: 300 }}>
          <TableBodyRow>
            <TableBodyCell>{note.created}</TableBodyCell>
            <TableBodyCell>{note.user}</TableBodyCell>
          </TableBodyRow>
        </div>
      </div>
    {/each}
  </TableBody>
</Table>
```

Then we end up having display issues with the row widths, and other problems (horizontal lines, etc):
<img width="918" alt="Screenshot 2024-07-23 at 1 13 02 PM" src="https://github.com/user-attachments/assets/d0ea56ea-e2fa-48a6-9fc3-31dccb12f9d7">

Is there any recommended workaround that could be employed to allow these transitions without messing up the table behavior?

**Comment by vettloffah** [comment]
You ever find a solution to this?

In regards to the layout issues you described in your example, I think it's also not proper HTML to have a `div` as a child of `tablebody` so that can create browser rendering issues

**Reply by brendan-morin** [reply]
No, I wasn't able to get this working, though with svelte 5 and the latest maybe there is a better story. I ended up moving off of flowbite tables in part for this reason.
